src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -308,14 +306,19 @@ protected virtual Task SetupRequest (HttpRequestMessage request, HttpURLConnecti
 		/// <param name="connection">HTTPS connection to return socket factory for</param>
 		protected virtual SSLSocketFactory? ConfigureCustomSSLSocketFactory (HttpsURLConnection connection)
 		{
-			return _underlyingHander.ConfigureCustomSSLSocketFactory (connection);
+			return _underlyingHander.ConfigureCustomSSLSocketFactoryInternal (connection);
 		}
 
 		[DynamicDependency (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof (AndroidMessageHandler))]
 		object GetUnderlyingHandler ()
 		{
 			var fieldName = "_nativeHandler";
-			var baseType = GetType ().BaseType;
+			var type = GetType ();
+			while (type != typeof (AndroidClientHandler)) {
```

Comment:
Does it make sense to also break when type == typeof(Object)?  Maybe throw as well when we don't have HttpClientHandler as the base type.

Reply:
This is a member function of `AndroidClientHandler` and so it will always be called on `AndroidClientHandler` or classes derived from it. I assume that this loop always has to stop but I can definitely throw an exception if for any reason we reach `object`.

Reply:
Might be a silly question, but do we also need to worry about never breaking out of the loop here?
I think having a throw if we reach `object` would be a good idea.

Reply:
If we reached the state where `type == typeof(object)`, then `type.BaseType` would return `null` and the next iteration would cause a null reference exception (not great but better than an infinite loop). I don't think it's possible to get to that state though as I described it the previous comment. Do you think there's some way to reach that state?